src/persistence/store/NodeFsJsonKeyValueStore.ts

```diff
@@ -0,0 +1,126 @@
+/**
+ * @license
+ * Copyright 2022 The node-matter Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import * as fs from "fs";
+import { JsonNoStorageMapKeyValueStore } from "../keyvalue/JsonKeyValueStore";
+import { Store } from "./Store";
+import { Time, Timer } from "../../time/Time";
+
+const STORAGE_DELAY_DEFAULT = 1000;
+
+interface StorageOptions {
+    /**
+     * Number of milliseconds to delay storing the data into the file to prevent storing too much when
+     * multiple data are set in a short time. Use -1 to only manually store the data by using "persistData".
+     * The default delay if not set is 1s
+     */
+    storageDelay?: number;
+}
+
+export class NodeFsJsonKeyValueStore extends JsonNoStorageMapKeyValueStore implements Store {
+    private opened = false;
+
+    private storageTimer: Timer | null = null;
+
+    constructor(
+        private readonly filePath: string,
+        private readonly options: StorageOptions = {}
+    ) {
+        super();
+        if (this.options.storageDelay === undefined || this.options.storageDelay < -1) {
+            this.options.storageDelay = STORAGE_DELAY_DEFAULT;
+        }
+    }
+
+    scheduleDataStorage(): void {
+        if (this.options.storageDelay === -1) {
+            return;
+        }
+
+        if (this.storageTimer) {
+            this.storageTimer.stop();
+            this.storageTimer = null;
+        }
```

Comment:
Doesn't this make it so you never persist if you write frequently? I suggest throttling (write at most every X time) instead of debouncing (write after no changes for X time).

Reply:
hmm ... good thought. Will change it